lib/index.js

```diff
@@ -0,0 +1,240 @@
+    electors,
+    // optional `f+1` "trusted" electors that must not fail and can handle
+    // catastrophic case where remaining `2f` electors cannot communicate
+    // (recovery electors are not a failsafe against a coordinated byzantine
+    // attack as the byzantine electors could collude to show different sides
+    // of a fork, but safe against network failures preventing consensus)
+    recoveryElectors,
+    // number of generations of merge events where no new electors participate
+    // and fewer than 2f+1 total electors (including self) participate causing
+    // recovery mode to trigger because consensus cannot be reached
+    recoveryGenerationThreshold,
+  };
+};
+
+// it is useful to override this function in tests
+api._computeElectors = async ({
+  blockHeight, electorPoolElectors, latestBlockSummary, ledgerConfiguration,
+  ledgerNode
+}) => {
+  // get partipicants for the last block
+
+  // FIXME: if we're using the _blocks API here, should it be made public
+  // (remove the underscore?)
+
+  const {_blocks} = ledgerNode.consensus;
+  const {consensusProofPeers, mergeEventPeers} = await _blocks.getParticipants(
+    {blockHeight: blockHeight - 1, ledgerNode});
+
+  // TODO: we should be able to easily remove previously detected
+  // byzantine nodes (e.g. those that forked at least) from the electors
+
+  // TODO: simply count consensus event signers once and proof signers
+  //   twice for now -- add comprehensive elector selection and
+  //   recommended elector vote aggregating algorithm in v2
+  const aggregate = {};
+  // add merge event peers
+  for(const id of mergeEventPeers) {
+    aggregate[id] = {id, weight: 1};
+  }
+
+  // TODO: weight previous electors more heavily to encourage continuity
+  // add consensusProof peers
+  for(const id of consensusProofPeers) {
+    if(id in aggregate) {
+      aggregate[id].weight = 3;
+    } else {
+      aggregate[id] = {id, weight: 2};
+    }
+  }
+  // add peers from the electorPool
+  for(const elector of electorPoolElectors) {
+    const electorType = [].concat(elector.type);
+    // electors defined in *any* way in the elector pool document should
+    // have the greatest weight
+    // FIXME: weight these like this?
+    // FIXME: is type === Continuity2017Peer a real thing?  If so, what do we
+    // do for that.
+    let weight = 100;
+    let recoveryElector = false;
+    if(electorType.includes(ELECTOR_TYPE_RECOVERY)) {
+      weight += 100;
+      recoveryElector = true;
+    }
+    if(electorType.includes(ELECTOR_TYPE_GUARANTOR)) {
+      weight += 100;
+    }
+    aggregate[elector.id] = {id: elector.id, recoveryElector, weight};
+  }
+
+  let electors = Object.values(aggregate);
+
+  let electorCount;
+  // the electorPool document may not be defined yet
+  if(electorPoolElectors.length > 0) {
+    // FIXME: `electorCount` in the ledgerConfiguration is used to specify a
+    // *maximum* number of electors correct?
```

Comment:
I recommend we put the max elector count in the elector pool document, not in the ledger configuration. That way we don't need to do a ledger config update to change it.

Reply:
But then how does a ledger that does not implement an electorPool make this specification?

Reply:
Well, maybe what we want to do is just treat an electorPool doc as an override for whatever is in the config. So, a config says "use this electorPool" unless there is no electorPool, in which case, use what's in the config. We already do that with the electors themselves, so let's apply the same rules to the limits.

Reply:
A TODO has been added to implement this feature in electorPool documents.  Resolved.